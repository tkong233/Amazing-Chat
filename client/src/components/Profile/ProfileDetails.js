import React, { useState } from 'react';
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import {
  Box,
  Button,
  Card,
  CardContent,
  CardHeader,
  Divider,
  makeStyles
} from '@material-ui/core';


import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MailOutlineIcon from '@material-ui/icons/MailOutline';
import EventIcon from '@material-ui/icons/Event';

const useStyles = makeStyles((theme) => ({
  root: {backgroundColor: 
    theme.palette.background.paper,}
}));

const ProfileDetails = ( props ) => {
  const classes = useStyles();
  const { date, email } = props.auth.user;
  return (
    <form
      autoComplete="off"
      noValidate
    >
      <Card>
        <CardHeader
          subheader="View your profile here"
          title="Profile"
        />
        <Divider />
        <CardContent>
        <div className={classes.root}>
        <ListItem >
          <ListItemIcon>
            <MailOutlineIcon />
          </ListItemIcon>
          <ListItemText primary={`Email: ${email}`} />
        </ListItem>
        <ListItem >
          <ListItemIcon>
            <EventIcon />
          </ListItemIcon>
          <ListItemText primary={`Registration date: ${date.substring(0, 10)}`} />
        </ListItem>
      </div></ CardContent>
        <Divider />
        <Box
          display="flex"
          justifyContent="space-between"
          p={2}
        >
          <Link to='/reset'>
          <Button
            color="primary"
            variant="contained"
          >
            Change Password
          </Button>
          </Link>

          <Button
            color="warning"
            variant="contained"
          >
            Deactivate the account
          </Button>
        </Box>
      </Card>
    </form>
  );
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
)(ProfileDetails);