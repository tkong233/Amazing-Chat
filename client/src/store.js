import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist'
import rootReducer from "./reducers";
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

const initialState = {};

const persistConfig = {
  key: 'root',
  storage,
}

const middleware = [thunk];

const persistedReducer = persistReducer(persistConfig, rootReducer, initialState);

const store = createStore(

  persistedReducer,
  compose(
    applyMiddleware(...middleware),
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
      compose
  )
);

const persistor = persistStore(store);

export {
  store,
  persistor,
  middleware
};



