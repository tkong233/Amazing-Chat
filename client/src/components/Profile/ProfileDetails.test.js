import React from 'react';
import ProfileDetails from './ProfileDetails';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import { applyMiddleware, createStore } from 'redux';
import {middleware} from '../../store';
import rootReducer from '../../reducers/index'

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

const setUp = (initialState={}) =>{
    
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    const store = createStoreWithMiddleware(rootReducer, initialState);
    const wrapper = shallow(<ProfileDetails store={store} />).dive();
    // console.log(wrapper.debug());
    return wrapper;
}

describe('Profile Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                updatePicture: ()=>{},
                logoutUser: () => {},
                auth: {user:{
                    date: '',
                    email: ''
                }},
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(ProfileDetails.propTypes, expectedProps, 'props', ProfileDetails.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const initialState = {
                auth: {user:{
                    date: '2020-01-01',
                    email: 'abcd@g.com'
                }}
            };
            wrapper = setUp(initialState);
        });

        it('Should render without errors', ()=>{
            const container = wrapper.find(`[data-test='ProfileDetailsComponent']`);
            expect(container.length).toBe(1);
        });

        it('Snapshot testing', ()=>{
            expect(wrapper).toMatchSnapshot();
        });

    });
});
