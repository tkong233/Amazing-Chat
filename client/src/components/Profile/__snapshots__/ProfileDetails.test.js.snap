// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
